header_guard.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+from pathlib import Path
+import argparse
+import re
+from typing import Optional, Tuple
+
+HEADER_EXTS: set[str] = {".h", ".hpp", ".hh", ".hxx", ".inl", ".tpp"}
+LEAD_RE = re.compile(r'^(?:\s*//[^\n]*\n|/\*.*?\*/\s*|\s*\n)+', re.DOTALL)
+PRAGMA_RE = re.compile(r'^\s*#\s*pragma\s+once\s*(?:\r?\n)?', re.IGNORECASE)
+GUARD_START_RE = re.compile(
+    r'^\s*#\s*ifndef\s+([A-Za-z_][A-Za-z0-9_]*)\s*$.*?^\s*#\s*define\s+\1\s*$',
+    re.MULTILINE | re.DOTALL,
+)
+END_RE = re.compile(r'^\s*#\s*endif\b.*(?:\r?\n)?', re.MULTILINE)
+
+
+def is_header_file(p: Path) -> bool:
+    return p.suffix.lower() in HEADER_EXTS
+
+
+def find_repo_root(start: Path) -> Path:
+    for d in [start] + list(start.parents):
+        if (d / ".git").exists():
+            return d
+    return Path(start.anchor or "/")
+
+
+def rel_to_root(root: Path, file: Path) -> Path:
```

Comment:
This functionname is not starting with a verb. Please change all function names to start with a verb.